lizmap/modules/lizmap/lib/Project/UserFilesFinder.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @author    3Liz
+ * @copyright 2024 3liz.com
+ *
+ * @see      3liz.com
+ *
+ * @license   MIT
+ */
+
+namespace Lizmap\Project;
+
+class UserFilesFinder
+{
+    public function listFileURLS(Project $project, $ignoreRepoAllowUserDefined = false)
+    {
+
+        $jsUrls = array();
+        $mjsUrls = array();
+        $cssUrls = array();
+        if ($ignoreRepoAllowUserDefined || $project->getRepository()->allowUserDefinedThemes()) {
+
+            $appContext = $project->getAppContext();
+            $jsDirArray = array('default', $project->getKey());
+            $repositoryPath = $project->getRepository()->getPath();
+            foreach ($jsDirArray as $dir) {
+
+                $items = array(
+                    // current repository
+                    'media/js/',
+                    // or root (js shared over repositories)
+                    '../media/js/',
+                );
+                foreach ($items as $item) {
+                    $jsPathRoot = realpath($repositoryPath.$item.$dir);
+                    if (is_dir($jsPathRoot)) {
```

Comment:
```suggestion
                    if (!is_dir($jsPathRoot)) {
```

To keep code on the left side of the editor, I find it more convenient negate, and a `continue` statement, so then all the code under can be pulled left.

Just a suggestion, about code visibility.

Reply:
i'm not very keen of using "continue" statement, let's other reviewers decide ?

Reply:
+1 for using continue.
This provides more readable content and avoids indentation nightmares

Reply:
This was not a strong opinion, just a way of coding that I prefer better.

Imagine a very long `if` block ... inside a loop. Having a `continue` ensures me that there is nothing under this very long `if` block. It puts some kind of safeguards that nothing will be executed under this statement

Just my 2 cents ;-)